src/psyclone/tests/domain/gocean/goloop_test.py

```diff
@@ -80,74 +80,94 @@ def test_goloop_no_children():
     gosched = GOInvokeSchedule('name', [])
-    gojloop = GOLoop(parent=gosched, loop_type="outer")
-    gosched.addchild(gojloop)
-    goiloop = GOLoop(parent=gojloop.loop_body, loop_type="inner")
-    gojloop.loop_body.addchild(goiloop)
-    gokern1 = GOKern()
-    gokern2 = GOKern()
-    # Set the index-offset of this kernel to a value that is not
-    # supported when using constant loop bounds
-    gokern1._index_offset = "go_offset_ne"
-    gokern2._index_offset = "go_offset_sw"
-    goiloop.loop_body.addchild(gokern1)
-    goiloop.loop_body.addchild(gokern2)
-    with pytest.raises(GenerationError) as excinfo:
-        goiloop.gen_code(None)
-    # Note that the kernels do not have a name, so there are empty quotes
-    assert "All Kernels must expect the same grid offset but kernel '' " \
-        "has offset 'go_offset_sw' which does not match 'go_offset_ne'" \
-        in str(excinfo.value)
+    goloop = GOLoop.create(parent=gosched,
+                           loop_type="inner",
+                           field_name="cv_fld",
+                           iteration_space="go_internal_pts",
+                           field_space="go_cv")
+
+    # Check the properties
+    assert isinstance(goloop, GOLoop)
+    assert goloop.loop_type == "inner"
+    assert goloop.field_name == "cv_fld"
+    assert goloop.iteration_space == "go_internal_pts"
+    assert goloop.field_space == "go_cv"
+
+    # Check that the created children correspond to the expected values
+    assert len(goloop.children) == 4
+    assert isinstance(goloop.children[0], Literal)
+    assert isinstance(goloop.children[1], Literal)
+    assert isinstance(goloop.children[2], Literal)
+    assert isinstance(goloop.children[3], Schedule)
+    assert goloop.children[0].value == '10'
+    assert goloop.children[1].value == '20'
+    assert goloop.children[2].value == '1'
+
+
+def test_goloop_properties_getters_and_setters():
+    ''' Test that the GOLoop getters and setters, retrieve and set the
+    expected attributes. '''
+    gosched = GOInvokeSchedule('name', [])
+    goloop = GOLoop(loop_type="inner", parent=gosched)
+
+    # Set and get iteration_space
+    goloop.iteration_space = "it_space"
+    assert goloop.iteration_space == "it_space"
+
+    # Set and get iteration_space
+    goloop.field_space = "cv_fld"
+    assert goloop.field_space == "cv_fld"
+
+    # Get bounds map
+    assert goloop.bounds_lookup == GOLoop._bounds_lookup
 
 
 def test_goloop_bounds_invalid_iteration_space():
     ''' Check that the _upper/lower_bound() methods raise the expected error
     if the iteration space is not recognised. '''
     gosched = GOInvokeSchedule('name', [])
     gojloop = GOLoop(parent=gosched, loop_type="outer")
-    # Have to turn-off constant loop bounds to get to the error condition
-    gosched._const_loop_bounds = False
+
     # Set the iteration space to something invalid
-    gojloop._iteration_space = "broken"
+    gojloop.iteration_space = "broken"
```

Comment:
Well.. I just added a str typecheck in the validation. I could add valid values but then we would force that custom boundaries are declared before setting the values on the PSyIR tree, and this is not the case at the moment.

So for instance the ordering of this statements would fail:
```
loop.iteration_space = "broken"
GOLoop.add_bounds("broken:go_ct:internal_we_halo:1:2:3:4")
```

Is it the desired behavior that this fails with this ordering?
(we also need to consider if "broken" is in the kernel metadata and if "broken" is specified in a config file to decide what are valid ordering. Now there are no restrictions.)

Reply:
Maybe we need to chat about this so that I fully understand what's going on, perhaps with @hiker  too since he has a better grip on the configuration stuff. I would have thought that it would be reasonable to assume that any configuration would be performed before we construct the PSyIR and thus we know what to expect. Am I missing something? I guess this would probably be a new Issue as it's not directly related to this one and can be a separate piece of work.

Reply:
See previous answer and #1393